### PR TITLE
fix(wallet-dashboard): sending max amount doesn't send the correct amount

### LINKED ISSUE #4229: Sending max amount doesn't send the correct amount
Issue only present in the dashboard

Repro:
1. Click the IOTA coin to send.
2. Click the `Max` Button
3. Click `Review`
4. Click Send
5. Wallet will open, `Balance Changes` will show something small like `-0.001 IOTA`

### PR DESCRIPTION
**Pull request by cpl121**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

Closes #4229 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by brancoder — APPROVED** [review]
This fixes the bug but the logic here seams a bit reversed to me. The `amount` holds the "formatted value" with decimal points and `formattedAmount` filed holds the full value without the decimal point. maybe this can be fixed in some other PR? wdyt?

**Review by brancoder — CHANGES_REQUESTED** [review]
Refactor the amount, formattedAmount and formatAmount:
https://github.com/iotaledger/iota/pull/4252#pullrequestreview-2464768916

**Review by VmMad — APPROVED** [review]

File: apps/wallet-dashboard/components/Dialogs/SendToken/SendTokenDialog.tsx [hunk 1]
```diff
@@ -33,19 +33,22 @@ function SendTokenDialogBody({
     const [step, setStep] = useState<FormStep>(FormStep.EnterValues);
     const [selectedCoin, setSelectedCoin] = useState<CoinBalance>(coin);
     const [formData, setFormData] = useState<FormDataValues>(INITIAL_VALUES);
+    const [fullAmount] = useFormatCoin(formData.amount, selectedCoin.coinType, CoinFormat.FULL);
```
Comment (panteleymonchuk) [inline apps/wallet-dashboard/components/Dialogs/SendToken/SendTokenDialog.tsx:36]: Is that ok that we don't show est gas fee in the new window?
image

For instance in the `develop` branch we show this value, but another point it's different from what we have in the dashboard
image

Comment (panteleymonchuk) [reply apps/wallet-dashboard/components/Dialogs/SendToken/SendTokenDialog.tsx:36]: btw, we have the same issue about different gas fee numbers for staking flow.

Here is a created issue.
https://github.com/iotaledger/iota/issues/4081

**Review by brancoder — APPROVED** [review]
LGTM 👍 

we could do something similar to wallet send token flow in wallet where we still have this unused code that you removed here:
```
const coinsIDs = [...coins]
            .sort((a, b) => Number(b.balance) - Number(a.balance))
            .map(({ coinObjectId }) => coinObjectId);
```
Should we do it in this PR or open anew one?

**Comment by brancoder** [comment]
> LGTM 👍
> 
> we could do something similar to wallet send token flow in wallet where we still have this unused code that you removed here:
> 
> ```
> const coinsIDs = [...coins]
>             .sort((a, b) => Number(b.balance) - Number(a.balance))
>             .map(({ coinObjectId }) => coinObjectId);
> ```
> 
> Should we do it in this PR or open anew one?

looks to me that coinIds is a leftover of old way of sending tx https://github.com/MystenLabs/sui/pull/8674/files

File: apps/wallet-dashboard/components/Dialogs/SendToken/SendTokenDialog.tsx (hunk 1 above)
Comment (cpl121) [reply apps/wallet-dashboard/components/Dialogs/SendToken/SendTokenDialog.tsx:36]: It works for me 🤔 Have you tried updating the wallet from this branch?

image

**Pull request merged**